Do not expose pg_trace.h (ie, <sys/sdt.h>) as something that every dependent
package will need to #include successfully.  Otherwise we'd need to make
postgresql-devel Require: systemtap-sdt-devel, which doesn't seem like a
desirable thing to do.  This patch is already applied upstream and won't
be needed in 8.3.7.


diff -Naur postgresql-8.3.6.orig/src/backend/access/transam/xact.c postgresql-8.3.6/src/backend/access/transam/xact.c
--- postgresql-8.3.6.orig/src/backend/access/transam/xact.c	2008-04-26 19:35:33.000000000 -0400
+++ postgresql-8.3.6/src/backend/access/transam/xact.c	2009-03-10 21:17:26.000000000 -0400
@@ -33,6 +33,7 @@
 #include "executor/spi.h"
 #include "libpq/be-fsstubs.h"
 #include "miscadmin.h"
+#include "pg_trace.h"
 #include "pgstat.h"
 #include "storage/fd.h"
 #include "storage/lmgr.h"
diff -Naur postgresql-8.3.6.orig/src/backend/storage/lmgr/lock.c postgresql-8.3.6/src/backend/storage/lmgr/lock.c
--- postgresql-8.3.6.orig/src/backend/storage/lmgr/lock.c	2008-03-04 14:54:13.000000000 -0500
+++ postgresql-8.3.6/src/backend/storage/lmgr/lock.c	2009-03-10 21:17:26.000000000 -0400
@@ -36,6 +36,7 @@
 #include "access/twophase.h"
 #include "access/twophase_rmgr.h"
 #include "miscadmin.h"
+#include "pg_trace.h"
 #include "pgstat.h"
 #include "utils/memutils.h"
 #include "utils/ps_status.h"
diff -Naur postgresql-8.3.6.orig/src/backend/storage/lmgr/lwlock.c postgresql-8.3.6/src/backend/storage/lmgr/lwlock.c
--- postgresql-8.3.6.orig/src/backend/storage/lmgr/lwlock.c	2008-01-01 14:45:52.000000000 -0500
+++ postgresql-8.3.6/src/backend/storage/lmgr/lwlock.c	2009-03-10 21:17:26.000000000 -0400
@@ -25,6 +25,7 @@
 #include "access/multixact.h"
 #include "access/subtrans.h"
 #include "miscadmin.h"
+#include "pg_trace.h"
 #include "storage/ipc.h"
 #include "storage/proc.h"
 #include "storage/spin.h"
diff -Naur postgresql-8.3.6.orig/src/include/c.h postgresql-8.3.6/src/include/c.h
--- postgresql-8.3.6.orig/src/include/c.h	2008-02-23 14:11:55.000000000 -0500
+++ postgresql-8.3.6/src/include/c.h	2009-03-10 21:17:26.000000000 -0400
@@ -57,7 +57,6 @@
 #include "pg_config_os.h"		/* must be before any system header files */
 #endif
 #include "postgres_ext.h"
-#include "pg_trace.h"
 
 #if _MSC_VER >= 1400
 #define errcode __msvc_errcode
